Blog Sample Data - New custom fields, tags

chmst opened the pull request:
### Summary of Changes
This PR adds new data to the content in blog sample data.

Tags
Fieldgroup
Field 

All loadLanguage statements come before the exception is thrown.

Modules are moved to position bottom-b


### Testing Instructions
Install blog sample data in a fresh installation.

In the backend you now must have 
a Field Group
a Field in this field group. 
four tags

In the frontend:
Check that articles have the new Tab "The Autor"
For two articles (Workflow and About your homepage) the author fields have some content.
In the frontend, the author - field must be shown below the text.

Tags are shown for articles in category blog

![grafik](https://user-images.githubusercontent.com/1035262/97818536-2a4f0200-1ca3-11eb-95c1-67a331bc8ac7.png)



### Documentation Changes Required
If there is a documentation about sample data: yes

chmst committed:
Add missing param secure to new menuItems

chmst committed:
Merge branch 'development' into sample-data-strike-again

chmst committed:
Add a Field Group for author information

chmst committed:
Fix modify and context

chmst committed:
Add a custom field to the field group

chmst committed:
Merge branch 'development' into sample-data-fields

chmst committed:
Add field values

chmst committed:
Fix field params

richard67 committed:
Merge branch 'development' into sample-data-fields

richard67 committed:
Merge branch 'development' into sample-data-fields

richard67 commented on plugins/sampledata/blog/blog.php:
```diff
@@ -119,6 +119,108 @@ public function onAjaxSampledataApplyStep1()
+		$groupTable->ordering        = 0;
+		$groupTable->created         = Factory::getDate()->toSql();
+		$groupTable->created_by      = $user->id;
+		$groupTable->modified        = Factory::getDate()->toSql();
+		$groupTable->modified_by     = $user->id;
```
```suggestion
		$groupTable->modified        = $groupTable->created;
		$groupTable->modified_by     = $groupTable->created_by;
```
When calling getDate 2 times, the times might not be exactly the same. But for something which never has been modified, the modification time has to be exactly the same as the creation time.

For consistency I do the same with the modified_by, even if there will not be a difference.

chmst replied:
Resolved in the new commit

richard67 reviewed (changes requested):
See suggested change above.

chmst committed:
Fix conditions for installation of custom fields and workflow

chmst commented:
Fixed setting the date in the field Group.

Fixed conditions for installing sample data with workflow and custom fields.

Workflow and/or custom fields can be disabled, but the articles and categories can be saved. 
Also a user could have the right to add fields but not worfklows.

chmst committed:
second try

richard67 commented:
Code review good, real test has to follow.

chmst committed:
use modelGroup for ssoting a group, fix message for duplicate field, remove useless language loads

chmst commented:
@richard67 Probably it is better to revert the changes in the core fields and add them there?

richard67 commented:
> @richard67 Probably it is better to revert the changes in the core fields and add them there?

You mean the changes in FieldTable.php and lib_joomla.ini?

Well, that seems to be indeed better in the CMS repo. But will it be possible to test it there when not using fields in blog sample data?

chmst commented:
Yes. So maybe you could make the test and then I revert changes in the field scripts?

richard67 commented:
Will test later tonight or tomorrow.

chmst committed:
revert changes which are better made in joomla-cms

chmst committed:
Add tags

chmst committed:
remove new step1, Integrate tags into step1, add language suffix to tags, load language before exception is thrown

chmst commented:
Tags are now integrated into the first step. The code would be clearer if the concerns are separated, but then we will have an issue with translation.
At the moment, there are 3 steps - more steps would require new language keys forthe  newly added steps.

chmst committed:
fix saving articles, fix comment

chmst committed:
resolve #155

richard67 commented:
I have tested this PR :white_check_mark: with success.

richard67 commented:
Thanks!
